src/servers/udp/server.rs

```diff
@@ -252,124 +329,178 @@ impl Udp {
         tx_start: oneshot::Sender<Started>,
         rx_halt: oneshot::Receiver<Halted>,
     ) {
-        let socket = Arc::new(
-            UdpSocket::bind(bind_to)
-                .await
-                .unwrap_or_else(|_| panic!("Could not bind to {bind_to}.")),
-        );
-        let address = socket
-            .local_addr()
-            .unwrap_or_else(|_| panic!("Could not get local_addr from {bind_to}."));
-        let halt = shutdown_signal_with_message(rx_halt, format!("Halting Http Service Bound to Socket: {address}"));
-
-        info!(target: "UDP TRACKER", "Starting on: udp://{}", address);
-
-        let running = tokio::task::spawn(async move {
-            debug!(target: "UDP TRACKER", "Started: Waiting for packets on socket address: udp://{address} ...");
-            Self::run_udp_server(tracker, socket).await;
-        });
+        let halt_task = tokio::task::spawn(shutdown_signal_with_message(
+            rx_halt,
+            format!("Halting Http Service Bound to Socket: {bind_to}"),
+        ));
+
+        let socket = tokio::time::timeout(Duration::from_millis(5000), Socket::new(bind_to))
+            .await
+            .expect("it should bind to the socket within five seconds");
+
+        let socket = match socket {
+            Ok(socket) => socket,
+            Err(e) => {
+                tracing::error!(target: "UDP TRACKER: Udp::run_with_graceful_shutdown", addr = %bind_to, err = %e, "panic! (error when building socket)" );
+                panic!("could not bind to socket!");
+            }
+        };
+
+        let address = socket.local_addr().expect("it should get the locally bound address");
+        let local_addr = format!("udp://{address}");
+
+        // note: this log message is parsed by our container. i.e:
+        //
+        // `[UDP TRACKER][INFO] Starting on: udp://`
+        //
+        tracing::info!(target: "UDP TRACKER", "Starting on: {local_addr}");
```

Comment:
Hi @da2ce7, maybe we could make this dependency explicit. We could move the const from:

- https://github.com/torrust/torrust-tracker/blob/develop/src/console/ci/e2e/logs_parser.rs#L5-L8
- https://github.com/torrust/torrust-tracker/blob/develop/src/console/ci/e2e/logs_parser.rs#L67-L69

to the UDP server mod. Ideally, we should also use a field in the log now that we are using tracing. But that could be implemented in a new PR.